src/builder/edit_message.rs

```diff
@@ -190,35 +187,40 @@ impl EditMessage {
         self
     }
 
-    /// Add a new attachment for the message.
+    /// Sets attachments, see [`EditAttachments`] for more details.
+    pub fn attachments(mut self, attachments: EditAttachments) -> Self {
+        self.attachments = Some(attachments);
+        self
+    }
+
+    /// Adds a new attachment to the message.
     ///
-    /// This can be called multiple times.
+    /// Resets existing attachments. See [`Self::keep_existing_attachment`] or read
+    /// [`EditAttachments`] for explanation.
     pub fn attachment(mut self, attachment: CreateAttachment) -> Self {
```

Comment:
```suggestion
    /// Resets existing attachments. See the documentation for [`EditAttachments`] for details.
    pub fn new_attachment(mut self, attachment: CreateAttachment) -> Self {
```
This method name isn't in line with the `new_attachment` method on `EditWebhookMessage`/`EditInteractionResponse`, and in combination with the new `attachments` method, might be confusing. I think it'd be better to change the name.